File: Server/src/main/java/Messages/CreateUserMessage.java
package Messages;

import javax.json.Json;
import javax.json.JsonObject;
import javax.json.JsonObjectBuilder;

/**
 * Created by Saharath Kleips on 11/12/2015.
 */
public class CreateUserMessage implements Message{
    static final String MESSAGE_TYPE = "CreateUser";
    static final String USERNAME_KEY = "Username";
    static final String PASSWORD_KEY = "Password";
    static final String CREATED_KEY = "Created";

    public String getUsername() { return username; }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public boolean isCreated() { return created; }

    public void setCreated(boolean created) {
        this.created = created;
    }

    String username;
    String password;
    boolean created;

    public CreateUserMessage(String username, String password, boolean created)
    {
        this.username = username;
        this.password = password;
        this.created = created;
    }

    public CreateUserMessage(JsonObject object)
    {
        JsonObject contents = JsonHelper.jsonFromString( object.get(MESSAGE_TYPE).toString() );
        this.username = contents.getString(USERNAME_KEY);
        this.password = contents.getString(PASSWORD_KEY);
        this.created = Boolean.parseBoolean( contents.get(CREATED_KEY).toString() );
    }

    public JsonObject toJson()
    {
        JsonObjectBuilder messageBuilder = Json.createObjectBuilder();
        JsonObjectBuilder thisBuilder = Json.createObjectBuilder();
        thisBuilder.add(USERNAME_KEY, this.username)
                   .add(PASSWORD_KEY, this.password)
                   .add(CREATED_KEY, this.created);

        messageBuilder.add(this.MESSAGE_TYPE, thisBuilder );
        return messageBuilder.build();
    }

    public byte[] getMessageBytes() { return toJson().toString().getBytes(); }

    public int getMessageLength() { return getMessageBytes().length; }
}

File: Clients/WindowsClient/WindowsClient/Messages/CreateUserMessage.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsClient.Messages
{
    class CreateUserMessage : Message
    {
        const String MESSAGE_TYPE = "CreateUser";
        const String USERNAME_KEY = "Username";
        const String PASSWORD_KEY = "Password";
        const String CREATED_KEY = "Created";
        
        public String Username
        {
            get { return username; }
            set { username = value; }
        }
        public String Password
        {
            get { return password; }
            set { password = value; }
        }
        public Boolean Created
        {
            get { return created; }
            set { created = value; }
        }

        String username;
        String password;
        Boolean created;

        public CreateUserMessage(String username, String password, Boolean created)
        {
            this.username = username;
            this.password = password;
            this.created = created;
        }

        public CreateUserMessage(JObject jobject)
        {
            JObject contents = JObject.Parse(jobject.GetValue(MESSAGE_TYPE).ToString());
            this.username = contents.GetValue(USERNAME_KEY).ToString();
            this.password = contents.GetValue(PASSWORD_KEY).ToString();
            this.created = Convert.ToBoolean(contents.GetValue(CREATED_KEY).ToString());
        }

        public JObject toJson()
        {
            JObject messageBuilder = new JObject();
            JObject thisBuilder = new JObject();
            thisBuilder.Add(USERNAME_KEY, this.username);
            thisBuilder.Add(PASSWORD_KEY, this.password);
            thisBuilder.Add(CREATED_KEY, this.created);

            messageBuilder.Add(MESSAGE_TYPE, thisBuilder);
            return messageBuilder;
        }

        public byte[] getMessageBytes()
        {
            ASCIIEncoding encoder = new ASCIIEncoding();
            return encoder.GetBytes(toJson().ToString(Formatting.None));
        }

        public int getMessageLength()
        {
            return getMessageBytes().Length;
        }
    }
}
